src/main/capi/config-c.cpp

```diff
@@ -10,14 +10,13 @@ duckdb_state duckdb_create_config(duckdb_config *out_config) {
 	if (!out_config) {
 		return DuckDBError;
 	}
-	DBConfig *config;
+	auto config = new DBConfig();
```

Comment:
Can we move this back into the `try/catch` block? Otherwise an allocation failure will result in an uncaught exception.

Reply:
Done 👍